#!/usr/bin/env python
PACKAGE = "laser_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Activate",     bool_t,      0,      "Activate GUI",                                 False)
gen.add("Invalid_r",    int_t,       0,      "Set segmented out points to this r value",     0,         0,  255)
gen.add("Invalid_g",    int_t,       0,      "Set segmented out points to this g value",     0,         0,  255)
gen.add("Invalid_b",    int_t,       0,      "Set segmented out points to this b value",     0,         0,  255)
gen.add("Invalid_x",    double_t,    0,      "Set segmented out points to this x value",     0.0,       0,  10)
gen.add("Invalid_y",    double_t,    0,      "Set segmented out points to this y value",     0.0,       0,  10)
gen.add("Invalid_z",    double_t,    0,      "Set segmented out points to this z value",     1.0,       0,  10)
gen.add("Leaf_Size",    double_t,    0,      "Set leaf size",                                0.01,      0,  1)
gen.add("Max_Iters",    int_t,       0,      "Euclidean cluster extraction max iters",       100,       0,  1000)
gen.add("Dist_Thresh",  double_t,    0,      "Euclidean cluster distance threshold",         0.01,      0,  1)
gen.add("Cluster_Tol",  double_t,    0,      "Euclidean cluster tolerance",                  0.02,      0,  1)
gen.add("Min_clus_pts", int_t,       0,      "Minumum cluster size",                         20,        0,  1000)
gen.add("Max_clus_pts", int_t,       0,      "Maximum cluster size",                         25000,     0,  100000)

exit(gen.generate(PACKAGE, "laser_detection", "CloudParams"))
